app/services/e_pubs_search_service.rb

```diff
@@ -28,14 +28,27 @@ def walk_up_dom(el, cfi, i)
     end
   end
 
```

Comment:
I spent too much time trying figuring out what all the letter variables stand for:

el = element
cfi = 
i = index
e = element
p0 = point0 a.k.a. lower bound a.k.a. left boundary
p1 = point1 a.k.a. upper bound a.k.a. right boundary
30 = offset before query and offset after query a.k.a. result window size a.k.a. PADDING = 30
q = query
stm = select from match
rs = resultset
els = elements

also, not sure exactly how find_hits works, but I know it is recursive,  looks like tails recursion, should I be concerned about stack overflow?

-Greg